src/Microsoft.Health.Dicom.Core/Features/Validation/StringElementValidation.cs

```diff
@@ -0,0 +1,31 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using EnsureThat;
+using FellowOakDicom;
+using FellowOakDicom.IO.Buffer;
+using Microsoft.Health.Dicom.Core.Extensions;
+
+namespace Microsoft.Health.Dicom.Core.Features.Validation;
+
+
+internal abstract class StringElementValidation : IElementValidation
+{
+    public void Validate(DicomElement dicomElement, ValidationLevel validationLevel = ValidationLevel.Default)
+    {
+        EnsureArg.IsNotNull(dicomElement, nameof(dicomElement));
+
+        string name = dicomElement.Tag.GetFriendlyName();
+        GetValue(dicomElement, out string value);
+        if (!string.IsNullOrEmpty(value) && validationLevel == ValidationLevel.Default)
+            value = value.TrimEnd('\0');
+
+        ValidateStringElement(name, value, dicomElement.ValueRepresentation, dicomElement.Buffer);
+    }
+
+    protected abstract void ValidateStringElement(string name, string value, DicomVR vr, IByteBuffer buffer);
+
+    protected abstract bool GetValue(DicomElement dicomElement, out string value);
```

Comment:
```suggestion
    protected abstract bool TryGetValue(DicomElement dicomElement, out string value);
```

Reply:
You can even make this `virtual` with a default implementation of:
```csharp
value = dicomElement.GetFirstValueOrDefault<string>();
return string.IsNullOrEmpty(value);
```

Reply:
The name was not changed

Reply:
Actually, you don't even use the return value from this method, so perhaps this method should just be called `GetValueOrDefault()` unless you want to incorporate the "valid" signal from this method too in your processing